packages/core/src/api/blockManipulation/moveBlock.test.ts

```diff
@@ -0,0 +1,194 @@
+import { afterAll, beforeAll, beforeEach, describe, expect, it } from "vitest";
+import { PartialBlock } from "../../blocks/defaultBlocks";
+import { BlockNoteEditor } from "../../editor/BlockNoteEditor";
+import { getBlockInfoFromPos } from "../getBlockInfoFromPos";
+import { NodeSelection, TextSelection } from "prosemirror-state";
+import { CellSelection } from "prosemirror-tables";
+import { moveBlockUp } from "./moveBlock";
+
+const blocks: PartialBlock[] = [
+  {
+    id: "paragraph-1",
+    type: "paragraph",
+    content: "Paragraph 1",
+    children: [
+      {
+        id: "nested-paragraph-1",
+        type: "paragraph",
+        content: "Nested Paragraph 1",
+      },
+    ],
+  },
+  {
+    id: "paragraph-2",
+    type: "paragraph",
+    content: "Paragraph 2",
+  },
+  {
+    id: "paragraph-3",
+    type: "paragraph",
+    content: "Paragraph 3",
+  },
+  {
+    id: "image-1",
+    type: "image",
+    props: {
+      url: "https://via.placeholder.com/150",
+    },
+  },
+  {
+    id: "table-1",
+    type: "table",
+    content: {
+      type: "tableContent",
+      rows: [
+        {
+          cells: ["Cell 1", "Cell 2", "Cell 3"],
+        },
+        {
+          cells: ["Cell 4", "Cell 5", "Cell 6"],
+        },
+        {
+          cells: ["Cell 7", "Cell 8", "Cell 9"],
+        },
+      ],
+    },
+  },
+  {
+    id: "trailing-paragraph",
+    type: "paragraph",
+  },
+];
+
+let editor: BlockNoteEditor;
+const div = document.createElement("div");
+
+function makeSelectionSpanContent(selectionType: "text" | "node" | "cell") {
+  const { startPos, contentNode } = getBlockInfoFromPos(
+    editor._tiptapEditor.state.doc,
+    editor._tiptapEditor.state.selection.from
+  );
+
+  if (selectionType === "cell") {
+    editor._tiptapEditor.view.dispatch(
+      editor._tiptapEditor.state.tr.setSelection(
+        CellSelection.create(
+          editor._tiptapEditor.state.doc,
+          editor._tiptapEditor.state.doc.resolve(startPos + 3).before(),
+          editor._tiptapEditor.state.doc
+            .resolve(startPos + contentNode.nodeSize - 3)
```

Comment:
fyi; at some point I want to get rid of all manual positions and keep these in helper functions; the reason is that they make implicit assumptions about our schema; so it's not future-proof when we start supporting more complex pm schemas

Reply:
Actually, can we do these positions more "semantically"? Like $from.start(), $from.end() for the text selectiontype